src/server.py

```diff
@@ -448,51 +447,28 @@ def run_seta_details_query():
 
     if buildbot:
         active_jobs = []
+        # pick up buildbot jobs from job list to faster the filter process
+        buildbot_jobs = [job for job in jobnames if job['buildplatform'] == 'buildbot']
+        # find out the correspond job detail information
+        for job in jobtype:
+            for j in buildbot_jobs:
+                if j['name'] == job[2] and j['platform'] == job[0] and j['buildtype'] == job[1]:
+                    active_jobs.append(j['ref_data_name'])
+        jobtype = active_jobs
+
+    if taskcluster:
+        active_jobs = []
+        taskcluster_jobs = [job for job in jobnames if job['buildplatform'] == 'taskcluster']
         for job in jobtype:
-            active_jobs.append(buildbot_name(job[0], job[1], job[2], branch))
+            for j in taskcluster_jobs:
+                if j['name'] == job[2] and j['platform'] == job[0] and j['buildtype'] == job[1]:
+                    active_jobs.append(j['ref_data_name'])
         jobtype = active_jobs
 
     retVal[date] = jobtype
     return {'jobtypes': retVal}
 
 
-# we still need this for builbot platform mapping
-def buildbot_name(platform, buildtype, jobname, branch):
-    platform_map = {}
-    # TODO: determine buildernames via alternative measures
-    platform_map['android-2-3-armv7-api9'] = "android-2-3-armv7-api9"
-    platform_map['android-4-2-x86'] = "android-4-2-x86"
-    platform_map['android-4-3-armv7-api11'] = "android-4-3-armv7-api11"
-    platform_map['android-4-3-armv7-api15'] = "android-4-3-armv7-api15"
-    platform_map['osx-10-10'] = "Rev7 MacOSX Yosemite 10.10.5"
-    platform_map['osx-10-8'] = "Rev5 MacOSX Mountain Lion 10.8"
-    platform_map['osx-10-6'] = "Rev4 MacOSX Snow Leopard 10.6"
-    platform_map['linux32'] = "Ubuntu VM 12.04"
-    platform_map['linux64'] = "Ubuntu VM 12.04 x64"
-    platform_map['linux64asan'] = "Ubuntu ASAN VM 12.04 x64"
-    platform_map['windowsxp'] = "Windows XP 32-bit"
-    platform_map['windows7-32'] = "Windows 7 32-bit"
-    platform_map['windows8-64'] = "Windows 8 64-bit"
-
-    buildtype_map = {}
-    buildtype_map["opt"] = "opt test"
-    buildtype_map["debug"] = "debug test"
-    buildtype_map["asan"] = "opt test"
-    buildtype_map["talos"] = "talos"
-
-    if buildtype == 'asan':
-        platform = 'linux64asan'
-
-    if not branch:
-        branch = "%s"
-
-    if jobname.startswith(('tp5o', 'chromez', 'dromaeojs', 'other', 'g1', 'g2', 'xperf', 'svgr')):
-        buildtype = 'talos'
-
-    # TODO: do we need to do a jobname conversion?  I don't see a need yet
-    return "%s %s %s %s" % (platform_map[platform], branch, buildtype_map[buildtype], jobname)
```

Comment:
to find all related jobs (similar, etc.) we need to have a matching platforms, buildtype, testtype values.  We had previously sanitized the platform names to be stuff like linux32, linux64, win32, win64.  Now we have buildbot values which are not guaranteed to be == to taskcluster values for platform.  It is ok to change the platform names in general, but we need some high degree of confidence that they won't change if somebody fiddles with a random config in-tree.

for the buildtype, this is important as well- right now we force things to opt/debug and treat asan as opt.  ideally we should be able to have opt/debug/asan- but then do we ignore other types like valgrind and pgo (I believe we should)?  If the names are not uniform between platforms or CI systems, then we need to force them to be similar by renaming them in SETA.